app/components/login/NewUserRegistration.js

```diff
@@ -0,0 +1,123 @@
+//Placeholder for newuser registration component
+
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateUser } from '../../redux/user';
+import { ErrorBar } from '../index';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { useStyles } from '../index';
+
+function NewUserRegistration(props) {
+  const [loginError, setError] = useState(false);
+  const user = useSelector(state => state.user);
+  const [initLoad, setInit] = useState(true);
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+
+  const classes = useStyles();
+
+  const handleChange = (event) => {
+    switch (event.target.id) {
+      case "firstname":
+        setFirstName(event.target.value);
+        break;
+      case "lastname":
+        setLastName(event.target.value);
+        break;
+        case "email":
+        setEmail(event.target.value);
+        break;
+        case "password":
+        setPassword(event.target.value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = () => {
+    setInit(false);
+    setError(false);
+    dispatch(updateUser(user.id, { firstName, lastName, email, password, userId: user.id }));
+    setEmail('');
+    setPassword('');
+    props.history.push(`/`)
+  }
+
+  useEffect(() => {
+    if (user && user.isRegistered) {
```

Comment:
this logic doesn't make sense -- a user shouldn't be able to get to this page via the UI if they're logged in. I would just redirect if user.isRegistered

Reply:
Done